### PR TITLE
HIVE-26269 Class cast exception when vectorization is enabled for cer…

### PR DESCRIPTION
**Pull request by ramesh0201**
…tain case when cases

### CODE REVIEW

**Comment by ramesh0201** [comment]
@scarlin-cloudera @maheshk114 @ayushtkn Can you please review the patch?

File: ql/src/test/queries/clientpositive/vector_case_when_3.q [hunk 1]
```diff
@@ -0,0 +1,7 @@
+set hive.explain.user=false;
+set hive.fetch.task.conversion=none;
+set hive.vectorized.execution.enabled=true;
+create external table test_decimal(rattag string, newclt_all decimal(15,2)) stored as orc;
+insert into test_decimal values('a', '10.20');
+select sum(case when rattag='a' then newclt_all*0.3 else newclt_all end) from test_decimal;
+select sum(case when rattag='Y' then newclt_all*0.3 else newclt_all end) from test_decimal;
```
Comment (abstractdog) [inline ql/src/test/queries/clientpositive/vector_case_when_3.q:7]: please include EXPLAIN VECTORIZATION DETAIL for the same query

Comment (ramesh0201) [reply ql/src/test/queries/clientpositive/vector_case_when_3.q:7]: Yes that will be useful to see what is happening. I added.

**Review by abstractdog — APPROVED** [review]
checked the patch locally, qtest fails without the fix, passes with it, LGTM

**Comment by ramesh0201** [comment]
Thanks for the review @abstractdog

**Pull request merged**